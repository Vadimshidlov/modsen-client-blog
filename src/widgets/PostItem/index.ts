export { PostItem } from '@/widgets/PostItem/ui/PostItem';
