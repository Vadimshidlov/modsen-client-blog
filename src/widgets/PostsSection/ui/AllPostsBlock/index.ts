export { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock/AllPostsBlock';
