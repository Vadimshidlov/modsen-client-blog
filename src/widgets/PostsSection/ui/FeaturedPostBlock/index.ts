export { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock/FeaturedPostBlock';
