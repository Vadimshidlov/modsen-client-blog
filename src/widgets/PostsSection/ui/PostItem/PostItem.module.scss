.post__container {
  padding: 32px 116px 25px 32px;
  display: flex;
  flex-direction: column;
  row-gap: 8px;
  cursor: pointer;
  transition: all .5s ease;

  @media (max-width: 768px) {
    padding: 16px 58px 12px 16px;
  }



  &:hover {
    background: $lightYellow;
  }

  .post__authorInfo {
    font-size: $fz16;
    font-weight: $regular;

    span {
      color: $lightFiolet;
    }

  }

  .post__text {
    max-width: 370px;
    font-size: $fz24;
    font-weight: $bold;
    line-height: 32px;

    @media (max-width: 768px) {
      font-size: $fz20;
    }

    @media (max-width: 450px) {
      font-size: $fz16;
    }
  }
}
