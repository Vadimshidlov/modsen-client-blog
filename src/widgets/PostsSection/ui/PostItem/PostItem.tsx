import React from 'react';
import styles from '@/widgets/PostsSection/ui/PostItem/PostItem.module.scss';
import { PostItemProps } from '@/shared/types';

export function PostItem({ postText, postDate, authorName }: PostItemProps) {
  return (
    <div className={styles.post__container}>
      <span className={styles.post__authorInfo}>
        By&nbsp;
        <span>{authorName}</span>
        &nbsp;
        {postDate}
      </span>
      <h4 className={styles.post__text}>{postText}</h4>
    </div>
  );
}
