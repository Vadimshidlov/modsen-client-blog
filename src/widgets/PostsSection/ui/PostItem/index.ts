export { PostItem } from '@/widgets/PostsSection/ui/PostItem/PostItem';
