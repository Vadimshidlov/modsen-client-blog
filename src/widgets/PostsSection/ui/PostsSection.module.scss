.posts {
  padding-top: 128px;
  margin-bottom: 128px;
  display: flex;
  justify-content: space-between;
  column-gap: 28px;

  @media (max-width: 1380px) {
    flex-direction: column;
    align-items: center;
  }

  @media (max-width: 768px) {
    padding-top: 64px;
    margin-bottom: 64px;
  }

  @media (max-width: 450px) {
    padding-top: 32px;
    margin-bottom: 32px;
  }
}
