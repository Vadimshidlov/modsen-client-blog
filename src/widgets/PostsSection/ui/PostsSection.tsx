import React from 'react';
import { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock';
import { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock';
import styles from '@/widgets/PostsSection/ui/PostsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function PostsSection() {
  return (
    <section className={`${styles.posts} ${container.container}`}>
      <FeaturedPostBlock />
      <AllPostsBlock />
    </section>
  );
}
