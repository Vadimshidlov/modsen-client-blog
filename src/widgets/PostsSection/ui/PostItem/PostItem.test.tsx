import React from 'react';
import { render } from '@testing-library/react';
import { PostItem } from '@/widgets/PostsSection/ui/PostItem';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('PostItem tests', () => {
  const postDate = '26 May, 2023';
  const authorName = 'John Doe';
  const postText = 'Post Text';

  test('should matches snapshot', () => {
    const { container } = render(
      <PostItem postDate={postDate} postText={postText} authorName={authorName} />,
    );
    expect(container).toMatchSnapshot();
  });
});
