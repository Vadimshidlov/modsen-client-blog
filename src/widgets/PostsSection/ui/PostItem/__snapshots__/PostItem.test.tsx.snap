// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem tests should matches snapshot 1`] = `
<div>
  <div
    class="post__container"
  >
    <span
      class="post__authorInfo"
    >
      By 
      <span>
        John Doe
      </span>
       
      26 May, 2023
    </span>
    <h4
      class="post__text"
    >
      Post Text
    </h4>
  </div>
</div>
`;
